sdk/language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -237,369 +237,416 @@ describe("decoders for recursive types do not loop", () => {
   });
 });
 
-test("create + fetch & exercise", async () => {
-  const aliceLedger = new Ledger({
-    token: ALICE_TOKEN,
-    httpBaseUrl: httpBaseUrl(),
-  });
-  const bobLedger = new Ledger({
-    token: BOB_TOKEN,
-    httpBaseUrl: httpBaseUrl(),
-  });
-  const aliceRawStream = aliceLedger.streamQuery(buildAndLint.Main.Person, {
-    party: ALICE_PARTY,
-  });
-  const aliceStream = promisifyStream(aliceRawStream);
-  // TODO(MH): Move this live marker into `promisifyStream`. Unfortunately,
-  // it didn't work the straightforward way and we need to spend more time
-  // figuring out what's going wrong before we can do it. There are two more
-  // instances of this pattern below.
-  const aliceStreamLive = pEvent(aliceRawStream, "live");
-  expect(await aliceStreamLive).toEqual([]);
-
-  const alice5: buildAndLint.Main.Person = {
-    name: "Alice from Wonderland",
-    party: ALICE_PARTY,
-    age: "5",
-    friends: [],
-  };
-  const bob4: buildAndLint.Main.Person = {
-    name: "Bob the Builder",
-    party: BOB_PARTY,
-    age: "4",
-    friends: [ALICE_PARTY],
-  };
-  const alice5Key = { _1: alice5.party, _2: alice5.age };
-  const bob4Key = { _1: bob4.party, _2: bob4.age };
-  const alice5Contract = await aliceLedger.create(
-    buildAndLint.Main.Person,
-    alice5,
-  );
-  expect(alice5Contract.payload).toEqual(alice5);
-  expect(alice5Contract.key).toEqual(alice5Key);
-  expect(await aliceStream.next()).toEqual([
-    [alice5Contract],
-    [{ created: alice5Contract, matchedQueries: [0] }],
-  ]);
-
-  let personContracts = await aliceLedger.query(buildAndLint.Main.Person);
-  expect(personContracts).toEqual([alice5Contract]);
-
-  const aliceContracts = await aliceLedger.query(buildAndLint.Main.Person, {
-    party: ALICE_PARTY,
-  });
-  expect(aliceContracts).toEqual(personContracts);
-  const bobContracts = await aliceLedger.query(buildAndLint.Main.Person, {
-    party: BOB_PARTY,
-  });
-  expect(bobContracts).toEqual([]);
+function doCreateFetchAndExercise<Pkg extends string>(
+  Person: Template<buildAndLint.Main.Person, buildAndLint.Main.Person.Key, Pkg>,
+  AllTypes: Template<buildAndLint.Main.AllTypes, undefined, Pkg>,
+  NonTopLevel: Template<buildAndLint.Lib.Mod.NonTopLevel, undefined, Pkg>,
```

Comment:
Pass in the descriptors of the templates we use, so that we can run it twice, once with the package id in the template ids, and once with the package name.